### PR TITLE
[move] Add minimum binary version

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Adds a minimum binary version to the deserializer, and plumbs this into the protocol config.

## Test plan 

Make sure existing tests pass, and also added a deserializer test to make sure we raise the correct error for a too-low binary version module. 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-open-rpc/spec/openrpc.json [hunk 1]
```diff
@@ -1847,6 +1847,7 @@
                 "max_verifier_meter_ticks_per_function": {
                   "u64": "6000000"
                 },
+                "min_move_binary_format_version": null,
```
Comment (cgswords) [inline crates/sui-open-rpc/spec/openrpc.json:1850]: I'm confused why this is `null`

File: crates/sui-types/src/execution_config_utils.rs [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+};
 use sui_protocol_config::ProtocolConfig;
 
 /// Build a `BinaryConfig` from a `ProtocolConfig`
 pub fn to_binary_config(protocol_config: &ProtocolConfig) -> BinaryConfig {
     BinaryConfig::new(
         protocol_config.move_binary_format_version(),
+        protocol_config
+            .min_move_binary_format_version_as_option()
+            .unwrap_or(VERSION_1),
```
Comment (tnowacki) [inline crates/sui-types/src/execution_config_utils.rs:16]: Was there a version 0? 
And this is the behavior on old code paths, correct?

**Review by tnowacki — APPROVED** [review]

**Review by dariorussi — APPROVED** [review]
it looks fine to me, incidentally this is doing nothing right? just exposing the min version

File: crates/sui-open-rpc/spec/openrpc.json (hunk 1 above)
Comment (tzakian) [reply crates/sui-open-rpc/spec/openrpc.json:1850]: This is the case for protocol version fields that are null/not used at previous versions (i.e., fields that have been added over time) is my understanding.

**Pull request merged**